src/main/scala/org/broadinstitute/dsde/firecloud/filematch/strategy/OntSingleReadStrategy.scala

```diff
@@ -0,0 +1,21 @@
+package org.broadinstitute.dsde.firecloud.filematch.strategy
+
+import org.broadinstitute.dsde.firecloud.filematch.result.{FailedMatchResult, FileMatchResult, SuccessfulMatchResult}
+import OntSingleReadStrategy.PATTERN
+
+import java.nio.file.{Path, Paths}
+import scala.util.matching.Regex
+
+class OntSingleReadStrategy extends FileRecognitionStrategy {
+
+  override def matchFirstFile(path: Path): FileMatchResult =
+    PATTERN.findFirstMatchIn(path.toString) match {
+      case Some(regexMatch) => SuccessfulMatchResult(path, Paths.get(""), regexMatch.group(1))
+      case None             => FailedMatchResult(path)
+    }
```

Comment:
Optional, but I just learned you could pattern match like this so I figured I'd share. It might obfuscate the fact that a regex is being used too much though, I haven't made up my mind.
```suggestion
    path.toString match {
      case PATTERN(id) => PartialMatchResult(path, id)
      case _           => FailedMatchResult(path)
    }
```

Reply:
Done: https://github.com/broadinstitute/firecloud-orchestration/pull/1581/commits/9ccf4944d496541fed79a497fdd1f31a24e50f05